src/lib/__tests__/ledger.test.ts

```diff
@@ -62,11 +59,30 @@ describe('ledger', () => {
     }
 
     const result = await faberAgent.ledger.getPublicDid(did);
-    expect(result).toEqual({
-      did: faberAgentPublicDid.did,
-      verkey: faberAgentPublicDid.verkey,
-      role: '101',
-    });
+
+    // TLDR: indy createAndStoreMyDid returns verkey "9iRCvQujJqjMDnX23kV64BKX2y9EzQdqLCHoFuge1apJ", ledger returns verkey "~Rn9y1NrwrLuHGYtCKDDRxN". Both are correct (did: GzKBNShs4KiBksKUkAZud2)
+    //
+    // .NET also abbreviates verkey before sending to ledger:
+    // https://github.com/hyperledger/aries-framework-dotnet/blob/f90eaf9db8548f6fc831abea917e906201755763/src/Hyperledger.Aries/Ledger/DefaultLedgerService.cs#L139-L147
+    //
+    //
+    // The verkey is 32 bytes, and by default in Indy the DID is chosen as the first 16 bytes of that key, before base58 encoding.
+    // The abbreviated verkey just replaces the first 16 bytes with ~ when it matches the DID.
+    //
+    // When full verkey is used to register on ledger, this is stored on ledger and also returned as full verkey
+    // same applies to abbreviated verkey. If it is used it will be stored abbreviated and returned abbreviated.
+    let verkey = faberAgentPublicDid.verkey as string;
+    if (isFullVerkey(verkey) && isAbbreviatedVerkey(result.verkey)) {
+      verkey = await indy.abbreviateVerkey(faberAgentPublicDid.did as string, verkey);
+    }
```

Comment:
What if we declared verkey type in `isFullVerkey` and `isAbbreviatedVerkey` function signature as `Verkey | string` would it help to get rid off these "overtyping" `as string`?

Reply:
The `as string` is because the `faberAgentPublicDid.did` can be `undefined` or `string` but as we set it explicitly some lines above we know it won't be `undefined`. 

But with the type improvements in the indy sdk I'm doing I think we can resolve this. The whole `faberAgentPublicDid` object will then be undefined and if it is set the `did` property will always be type `string`. Does that make sense?

Reply:
Typescript knows `Verkey` type is the same as `string` so it won't complain about the differnce

Reply:
> Typescript knows `Verkey` type is the same as `string` so it won't complain about the differnce

Good, that's what I would expect, but I thought maybe TS is not so great 😄 

I'm thinking if the `if` check and throwing error would not be better for understanding. Of course, it's not better from the boilerplating perspective, but I was confused by that `as string` a little bit and if someone decides to move `faberAgentPublicDid = faberAgent.getPublicDid();`  to `beforeAll` and remove the test we're in trouble :)

Not saying this is bad and we'll solve it by your work with types anyway, just sharing my perspective for such situations in general ;)